src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs

```diff
@@ -788,4 +788,23 @@ public async Task RequestDelegatePopulatesFromOptionalFormParameterStringArray()
 
         Assert.Equal<string[]>(["hello", "bye"], (string[])httpContext.Items["message"]);
     }
+
+    [Theory]
+    [InlineData("""app.MapGet("/", ([FromHeader(Name = "q")] string[]? arr) => arr);""", "",      "[]")]
+    [InlineData("""app.MapGet("/", ([FromHeader(Name = "q")] string[]? arr) => arr);""", "a,b,c", "[\"a\",\"b\",\"c\"]")]
+    [InlineData("""app.MapGet("/", ([FromHeader(Name = "q")] int[]? arr) => arr);""",    "1,2,3", "[1,2,3]")]
+    public async Task RequestDelegateGenerator_FromHeader_CommaSeparatedValues(string source, string headerContent, string expectedBody)
```

Comment:
If you move this test case to `RequestDelegateCreationTests.Arrays.cs` we'll get coverage for both compile-time and run-time code gen with a single test case.

Reply:
@captainsafia 
moved